lxd/patches.go

```diff
@@ -596,42 +549,70 @@ func upgradeFromStorageTypeLvm(name string, d *Daemon, defaultPoolName string, d
 
 	// Get size of the volume group.
 	output, err := tryExec("vgs", "--nosuffix", "--units", "g", "--noheadings", "-o", "size", defaultPoolName)
-	if err != nil {
-		return err
-	}
-	tmp := string(output)
-	tmp = strings.TrimSpace(tmp)
-	szFloat, err := strconv.ParseFloat(tmp, 32)
-	if err != nil {
-		return err
-	}
-	szInt64 := shared.Round(szFloat)
-	poolConfig["size"] = fmt.Sprintf("%dGB", szInt64)
+	if err == nil {
+		tmp := string(output)
+		tmp = strings.TrimSpace(tmp)
+		szFloat, err := strconv.ParseFloat(tmp, 32)
+		if err != nil {
+			return err
+		}
+		szInt64 := shared.Round(szFloat)
+		poolConfig["size"] = fmt.Sprintf("%dGB", szInt64)
 
-	err = storagePoolValidateConfig(defaultPoolName, "lvm", poolConfig)
-	if err != nil {
-		return err
+		err = storagePoolValidateConfig(defaultPoolName, "lvm", poolConfig)
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
We shouldn't just move on if the LVM is broken to the point where we can't even list the VGs...